.github/workflows/common-post-jobs.yaml

```
@@ -64,3 +64,9 @@ jobs:
           sha: ${{ inputs.sha }}
           status: ${{ github.event_name != 'schedule' && 'success' || 'failure' }}
           description: 'Skipped'
+
+      - name: Fail job if result is failure
```

Nit. I'd suggest adding a comment why this is necessary (basically what you wrote in the commit message), just to avoid that it gets removed in the future.

I'd also suggest to move it to a separate job, just to make it clearer that it is expected to fail upon other failures, and not hide possible failures of the `commit-status-final` one.

---

Makes sense! Done.